Terminal.Gui/Drawing/Gradient.cs

```diff
@@ -0,0 +1,243 @@
+﻿// This code is a C# port from python library Terminal Text Effects  https://github.com/ChrisBuilds/terminaltexteffects/
+
+namespace Terminal.Gui;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// Describes the pattern that a <see cref="Gradient"/> results in e.g. <see cref="Vertical"/>, <see cref="Horizontal"/> etc
+/// </summary>
+public enum GradientDirection
+{
+    /// <summary>
+    /// Color varies along Y axis but is constant on X axis.
+    /// </summary>
+    Vertical,
+
+    /// <summary>
+    /// Color varies along X axis but is constant on Y axis.
+    /// </summary>
+    Horizontal,
+
```

Comment:
Please use re-sharper's Clean Up Code for all these files.

![image](https://github.com/gui-cs/Terminal.Gui/assets/585482/bf63e784-db79-42bd-b96d-2420d0513fdb)

![image](https://github.com/gui-cs/Terminal.Gui/assets/585482/054f2a76-e91f-4503-967d-e5b461da4c51)

Reply:
I will check when I have access to my windows machine but don't think I have a license for this tool

Reply:
I have licenses for the team. I'll send.